homeassistant/components/litterrobot/switch.py

```diff
@@ -17,34 +17,47 @@
 from .entity import LitterRobotEntity, _RobotT
 
 
-@dataclass(frozen=True)
-class RequiredKeysMixin(Generic[_RobotT]):
-    """A class that describes robot switch entity required keys."""
-
-    set_fn: Callable[[_RobotT, bool], Coroutine[Any, Any, bool]]
-
-
-@dataclass(frozen=True)
-class RobotSwitchEntityDescription(SwitchEntityDescription, RequiredKeysMixin[_RobotT]):
+@dataclass(frozen=True, kw_only=True)
+class RobotSwitchEntityDescription(SwitchEntityDescription, Generic[_RobotT]):
     """A class that describes robot switch entities."""
 
     entity_category: EntityCategory = EntityCategory.CONFIG
+    set_fn: Callable[[_RobotT, bool], Coroutine[Any, Any, bool]]
+    value_fn: Callable[[_RobotT], bool]
 
 
 ROBOT_SWITCHES = [
     RobotSwitchEntityDescription[LitterRobot | FeederRobot](
         key="night_light_mode_enabled",
         translation_key="night_light_mode",
         set_fn=lambda robot, value: robot.set_night_light(value),
+        value_fn=lambda robot: robot.night_light_mode_enabled,
     ),
     RobotSwitchEntityDescription[LitterRobot | FeederRobot](
         key="panel_lock_enabled",
         translation_key="panel_lockout",
         set_fn=lambda robot, value: robot.set_panel_lockout(value),
+        value_fn=lambda robot: robot.panel_lock_enabled,
     ),
 ]
 
 
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: LitterRobotConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Litter-Robot switches using config entry."""
+    hub = entry.runtime_data
+    entities = [
+        RobotSwitchEntity(robot=robot, hub=hub, description=description)
+        for description in ROBOT_SWITCHES
+        for robot in hub.account.robots
+        if isinstance(robot, (LitterRobot, FeederRobot))
+    ]
+    async_add_entities(entities)
```

Comment:
`async_add_entities` will accept a generator expression so you could write it as

```
async_add_entities(
    RobotSwitchEntity(robot=robot, hub=hub, description=description)
    for description in ROBOT_SWITCHES
    for robot in hub.account.robots
    if isinstance(robot, (LitterRobot, FeederRobot))
)
```

But it might be a tiny bit slower since list comps are inlined in Python 3.12+

Reply:
There is a mix of both list comp and generators in this integration. Should I update this to be a generator or should I switch other files to use list comp?

Reply:
Generally I prefer the generator expression since its more compact and this code isn't called frequently enough that performance is a consideration.  I'm neutral on the topic, and fine merging as is if you prefer.

Reply:
I'll leave it as is then and address consistency in a future PR as I work to >=bronze quality scale